boa/src/builtins/object/gcobject.rs

```diff
@@ -61,6 +65,55 @@ impl Display for BorrowError {
     }
 }
 
+struct RecursionLimiter {
+    free: bool,
+    first: bool,
+}
```

Comment:
Can we add some documentation here? And derive `Debug`, `Clone` and `Copy`, just in case.

Reply:
`Copy` can't be implemented because `RecursionLimiter` implements `Drop`.